client/components/email-verification/email-verification-dialog/index.jsx

```diff
@@ -0,0 +1,118 @@
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import noop from 'lodash/noop';
+import { localize } from 'i18n-calypso';
+import { connect } from 'react-redux';
+import { get, includes } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import Dialog from 'components/dialog';
+import Button from 'components/button';
+import Spinner from 'components/spinner';
+import { getCurrentUserEmail } from 'state/current-user/selectors';
+import {
+	verifyEmail,
+	resetVerifyEmailState,
+} from 'state/current-user/email-verification/actions';
+
+class VerifyEmailDialog extends Component {
+
+	getResendButtonLabel() {
+		if ( 'sent' === this.props.emailVerification ||
+			'error' === this.props.emailVerification ) {
+			return this.props.translate( 'Email Sent' );
+		}
+		if ( 'requesting' === this.props.emailVerification ) {
+			return <Spinner className="email-verification-dialog__confirmation-dialog-spinner" />;
+		}
+		return this.props.translate( 'Resend Email' );
+	}
+
+	handleClose = () => {
+		this.props.resetVerifyEmailState();
+		this.props.onClose();
+	};
+
+	getDialogButtons() {
+		return [
+			<Button
+				key="resend"
+				primary={ false }
+				disabled={ includes( [ 'requesting', 'sent', 'error' ], this.props.emailVerification ) }
+				onClick={ this.props.verifyEmail }>
+				{ this.getResendButtonLabel() }
+			</Button>,
+			<Button
+				key="close"
+				primary={ true }
+				onClick={ this.handleClose }>
+					{ this.props.translate( 'OK' ) }
+			</Button>
+		];
+	}
+
+	render() {
+		const strings = {
+			confirmHeading: this.props.translate( 'Please confirm your email address' ),
+
+			confirmExplanation: this.props.translate( 'We sent you an email when you first signed up. ' +
+				'Please open the message and click the blue button.' ),
+
+			confirmReasoning: this.props.translate( 'Email confirmation allows us to assist when recovering ' +
+				'your account in the event you forget your password.' ),
+
+			confirmEmail: this.props.translate(
+				'{{wrapper}}%(email)s{{/wrapper}} {{emailPreferences}}change{{/emailPreferences}}',
+				{
+					components: {
+						wrapper: <span className="email-verification-dialog__confirmation-dialog-email-wrapper" />,
+						emailPreferences: <a href="/me/account" />
+					},
+					args: {
+						email: this.props.email
+					}
+				}
+			)
+		};
```

Comment:
Yeh, I thought about it, but there were a few other functions where it didn't make sense to destructure (i.e. just for using it once or twice), so I left all the uses the same throughout for consistency.